src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,501 @@
+}
+
+impl BreadcrumbModel {
+    /// Constructor.
+    pub fn new<'t,S:Into<&'t Scene>>
+    (scene:S, frp:&Frp,method_pointer:&MethodPointer, expression_id:&ast::Id) -> Self {
+        let scene             = scene.into();
+        let logger            = Logger::new("Breadcrumbs");
+        let display_object    = display::object::Instance::new(&logger);
+        let view_logger       = Logger::sub(&logger,"view_logger");
+        let view              = component::ShapeView::<background::Shape>::new(&view_logger, scene);
+        let icon              = component::ShapeView::<icon::Shape>::new(&view_logger, scene);
+        let separator         = component::ShapeView::<separator::Shape>::new(&view_logger, scene);
+        let font              = scene.fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let glyph_system      = GlyphSystem::new(scene,font);
+        let label             = glyph_system.new_line();
+        let expression_id     = *expression_id;
+        let method_pointer    = method_pointer.clone();
+        let info              = Rc::new(BreadcrumbInfo{method_pointer,expression_id});
+        let animations        = Animations::new(&frp.network);
+        let is_selected       = default();
+        let relative_position = default();
+
+        let shape_system = scene.shapes.shape_system(PhantomData::<background::Shape>);
+        scene.views.main.remove(&shape_system.shape_system.symbol);
+        scene.views.breadcrumbs.add(&shape_system.shape_system.symbol);
+
+        let shape_system = scene.shapes.shape_system(PhantomData::<icon::Shape>);
+        shape_system.shape_system.set_pointer_events(false);
+        scene.views.main.remove(&shape_system.shape_system.symbol);
+        scene.views.breadcrumbs.add(&shape_system.shape_system.symbol);
+
+        let shape_system = scene.shapes.shape_system(PhantomData::<separator::Shape>);
+        shape_system.shape_system.set_pointer_events(false);
+        scene.views.main.remove(&shape_system.shape_system.symbol);
+        scene.views.breadcrumbs.add(&shape_system.shape_system.symbol);
+
+        let symbol = glyph_system.sprite_system().symbol();
+        scene.views.main.remove(&symbol);
+        scene.views.breadcrumbs.add(&symbol);
+
+        Self{logger,view,icon,separator,display_object,glyph_system,label,info,animations
+            ,is_selected,relative_position}.init()
+    }
+
+    fn init(self) -> Self {
+        self.add_child(&self.view);
+        self.view.add_child(&self.separator);
+        self.separator.add_child(&self.icon);
+        self.icon.add_child(&self.label);
+
+        let color  = if self.is_selected() { FULL_COLOR } else { TRANSPARENT_COLOR };
+
+        self.label.set_font_size(TEXT_SIZE);
+        self.label.set_font_color(color);
+        self.label.set_text(&self.info.method_pointer.name);
+        self.label.set_position(Vector3(ICON_RADIUS + ICON_RIGHT_MARGIN, -TEXT_SIZE/3.0, 0.0));
```

Comment:
`TEXT_SIZE/3.0` is a hardcoded constant - why its hardcoded this way? Where the 3.0 comes from ?